include/NovelRT/Exceptions/OpenGLLinkageFailure.h

```diff
@@ -0,0 +1,19 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_OPENGL_LINKAGE_FAILURE_EXCEPTION_H
+#define NOVELRT_OPENGL_LINKAGE_FAILURE_EXCEPTION_H
+
+#include <stdexcept>
+#include <string>
+
+namespace NovelRT::Exceptions {
+  class OpenGLLinkageFailure final : public std::runtime_error {
+  public:
+    OpenGLLinkageFailure(unsigned int programId, const std::string &errorMessage) :
```

Comment:
The reference symbols should be next to the type, not the parameter name. also prefer `uint32_t` over `unsigned int`.